yo1k/sandbox/exercises/CSC_algorithms/test_sorting.py

```diff
@@ -0,0 +1,54 @@
+import itertools
+from collections.abc import Sequence, MutableSequence
+from typing import Any
+
+import pytest
+
+import yo1k.sandbox.exercises.CSC_algorithms.sorting as my_sort
+
+
+@pytest.fixture
+def arrays_set() -> Sequence[MutableSequence[Any]]:
+    return (
+            [1, 2, 5, 7, 3, 10, 4, -1, 5],
+            [1, 2, 3, 2, 7, 8, 4, 5, 2],
+            [1, -2, -3, -2, -7, -8, -4, -5, -2],
+            [2, 2, 2, 2, 2],
+            [1, 2, 3, 4, 5],
+            [5, 4, 3, 2, 1],
+            [3, 3, 1, 2, 1, 2, 1],
+            [-3, 1],
+            [-1],
+            [])
+
+
+def test_swap() -> None:
+    assert my_sort.swap([1, 2, 5, 7, 3, 10, 4, -1, 5], 2, 4) == [1, 2, 3, 7, 5, 10, 4, -1, 5]
+
+
+def test_selection_sort(arrays_set: Sequence[MutableSequence[Any]]) -> None:
+    for array in arrays_set:
+        array1, array2 = itertools.tee(array)
+        assert my_sort.selection_sort(list(array1)) == sorted(list(array2))
```

Comment:
The documentation says: [_"In general, if one iterator uses most or all of the data before another iterator starts, it is faster to use `list()` instead of `tee()`."_](https://docs.python.org/3/library/itertools.html#:~:text=In%20general%2C%20if%20one%20iterator%20uses%20most%20or%20all%20of%20the%20data%20before%20another%20iterator%20starts%2C%20it%20is%20faster%20to%20use%20list()%20instead%20of%20tee().) This seems like you case. Why do you use `tee` nonetheless here and in other tests, and then use `list` on top of that?

Reply:
Done.

I used `tee()` to create 2 copies of the `array` and in this case I needed exactly `tee()`, not `list()`. I used `list(array1)` (`list(array2)`) to create a list because `array1` (`array2`) is a iterator.
I corrected this part of code using [`copy.deepcopy()`](https://docs.python.org/3/library/copy.html#:~:text=copy.deepcopy(x%5B%2C%20memo%5D)%C2%B6).